internal/pkg/controllers/profile/profile.go

```diff
@@ -223,6 +257,79 @@ func (r *Reconciler) reconcileSeccompProfile(
 	return reconcile.Result{}, nil
 }
 
+// retryAddFinalizer adds a finalizer for the current node to the resource, retrying in case other pods in the
+// operator DaemonSet are actively doing the same thing and causing conflicts.
+func (r *Reconciler) retryAddFinalizer(ctx context.Context, sp *v1alpha1.SeccompProfile, finalizer string) error {
+	retries := 5
+	namespacedName := types.NamespacedName{Name: sp.ObjectMeta.Name, Namespace: sp.ObjectMeta.Namespace}
+	var err error
+	for i := 0; i < retries; i++ {
+		// Refresh object
+		if err := r.client.Get(ctx, namespacedName, sp); err != nil {
+			return errors.Wrap(err, "retry adding finalizer")
+		}
+		if controllerutil.ContainsFinalizer(sp, finalizer) {
+			return nil
+		}
+		controllerutil.AddFinalizer(sp, finalizer)
+		err = r.client.Update(ctx, sp)
+		if err == nil {
+			return nil
+		}
+		if kerrors.IsConflict(err) {
+			continue
+		}
+		if err != nil {
+			return errors.Wrap(err, "retry adding finalizer")
+		}
+	}
+	return errors.Wrap(err, fmt.Sprintf("exceeded %d retries attempting to add finalizer to seccompprofile", retries))
+}
+
+// retryRemoveFinalizer removes a finalizer for the current node to the resource, retrying in case other pods in the
+// operator DaemonSet are actively doing the same thing and causing conflicts.
+func (r *Reconciler) retryRemoveFinalizer(ctx context.Context, sp *v1alpha1.SeccompProfile, finalizer string) error {
+	retries := 5
+	namespacedName := types.NamespacedName{Name: sp.ObjectMeta.Name, Namespace: sp.ObjectMeta.Namespace}
+	var err error
+	for i := 0; i < retries; i++ {
+		fmt.Printf("trying to remove finalizer %s\n", finalizer)
+		// Refresh object
+		if err := r.client.Get(ctx, namespacedName, sp); err != nil {
+			return errors.Wrap(err, "retry removing finalizer")
+		}
+		if !controllerutil.ContainsFinalizer(sp, finalizer) {
+			return nil
+		}
+		controllerutil.RemoveFinalizer(sp, finalizer)
+		err = r.client.Update(ctx, sp)
+		if err == nil {
+			return nil
+		}
+		if kerrors.IsConflict(err) {
+			continue
+		}
+		if err != nil {
+			return errors.Wrap(err, "retry removing finalizer")
+		}
+	}
+	return errors.Wrap(err, fmt.Sprintf("exceeded %d retries removing finalizer from seccompprofile", retries))
+}
+
+func handleDeletion(sp *v1alpha1.SeccompProfile, l logr.Logger) error {
+	profilePath, err := GetProfilePath(sp.ObjectMeta.Name, sp.ObjectMeta.Namespace, sp.Spec.TargetWorkload)
+	if err != nil {
+		return err
+	}
+	if _, e := os.Stat(profilePath); !os.IsNotExist(e) {
```

Comment:
I'm wondering why we `Stat` here before the `Remove`. I think remove does return `os.IsNotExist(e)` too in case that the file does not exist.

Reply:
fixed